pkg/validation/internal/bundle.go

```diff
@@ -166,6 +166,18 @@ func validateBundleSize(bundle *manifests.Bundle) []errors.Error {
 			bundle.Name))
 	}
 
+	// Before these versions the bundles were not compressed
+	// and their size must be < ~1MB
+	// todo: remove me at 2023
+	if bundle.Size > max_bundle_size {
+		errs = append(errs, errors.WarnInvalidBundle(
+			fmt.Sprintf("bundle uncompressed size exceeded the limit support for OLM versions relesed prior 1.17.5 :  size=~%s , max=%s. "+
+				"(these bundle cannot work in any cluster or vendor which uses OLM versions < 1.17.5 like OpenShift versions < 4.9)",
+				formatBytesInUnit(bundle.Size),
+				formatBytesInUnit(max_bundle_size)),
+			bundle.Name))
+	}
+
```

Comment:
From @joelanford : 

> I feel somewhat strongly that we should keep the API repo focused on just one level of compatibility with OPM/OLM, so I'd suggest we don't check the uncompressed size at all since current versions of OPM/OLM do compression.

> Otherwise, it seems like the API repo's main branch will start growing to encompass validation for all past history of what OPM/OLM support.
I feel somewhat strongly that we should keep the API repo focused on just one level of compatibility with OPM/OLM, so I'd suggest we don't check the uncompressed size at all since current versions of OPM/OLM do compression.

Otherwise, it seems like the API repo's main branch will start growing to encompass validation for all past history of what OPM/OLM support.

Reply:
I agree with your thought in general. But maybe is valid to think a little on this one because the change to compress which allows large bundles did not happen a long time ago and many authors publish their solution without being aware of this limitation. 

However, a strong con is we begin to provide expectations that we will begin to write code for old versions as you said. That is a really good reason for we drop it out.

Reply:
/hold 

to think

Reply:
@joelanford, 

WDYT? I tried to clarify the con raised by you and address this one as well. 
/hold cancel

Reply:
Isn't it kind of the same problem in other areas of Kubernetes? v1beta1 CRDs were removed in 1.22, so we went and removed all the code that supported v1beta1 CRDs in controller-runtime and controller-tools. If someone still needs to do something with v1beta1 CRDs we'd suggest using older versions of those tools.

IMO the same thing applies here. I'm guessing the friction is that we only use one version of the validator across all the versions of OLM that we support. Shouldn't we have a validator version per OLM version such that when we're validating a bundle for a particular version of OLM we use the validator version that goes with that OLM version?

Reply:
We still parsing and allowing the v1beta1 in order to support old k8s versions.
However, fair enough. I really understand where you came from and I can go along with it.